R/zchunk_LA161.Cstorage_china.R

```diff
@@ -0,0 +1,125 @@
+#' module_gcam.china_LA161.Cstorage
+#'
+#' Calculates onshore CO2 storage by grid region.
+#'
+#' @param command API command to execute
+#' @param ... other optional parameters, depending on command
+#' @return Depends on \code{command}: either a vector of required inputs,
+#' a vector of output names, or (if \code{command} is "MAKE") all
+#' the generated outputs: \code{L161.Cstorage_province}. The corresponding file in the
+#' original data system was \code{LA161.Cstorage.R} (gcam-china level1).
+#' @details Calculates onshore CO2 storage by grid region.
+#' @importFrom assertthat assert_that
+#' @importFrom dplyr filter mutate select
+#' @importFrom tidyr gather spread
+#' @author Liu August 2018
+
+module_gcam.china_LA161.Cstorage <- function(command, ...) {
+  if(command == driver.DECLARE_INPUTS) {
+    return(c(FILE = "gcam-china/province_names_mappings",
+             FILE = "gcam-china/CCS_provincial_data",
+             FILE = "gcam-china/CCS_provincial_data_add",
+             FILE = "gcam-china/unallocated_CStorage_province"))
+  } else if(command == driver.DECLARE_OUTPUTS) {
+    return(c("L161.Cstorage_province"))
+  } else if(command == driver.MAKE) {
+
+    # Silence package checks
+    province <- grid_region <- Cost_2005USDtCO2 <- CO2_Mt <- MtC <- Cumul_MtC <- quantile <-
+      grade <- Cost_1990USDtC <- NULL
+
+    all_data <- list(...)[[1]]
+
+    # -----------------------------------------------------------------------------
+    # 1.Load required inputs
+    province_names_mappings <- get_data(all_data, "gcam-china/province_names_mappings")
+    Dooley_CCS_China        <- get_data(all_data, "gcam-china/CCS_provincial_data")
+    CCS_China_add           <- get_data(all_data, "gcam-china/CCS_provincial_data_add")
+    Unallocated_CStorage    <- get_data(all_data, "gcam-china/unallocated_CStorage_province")
+
+    # -----------------------------------------------------------------------------
+    # 2.perform computations
+    L161.Csupply_province <- Dooley_CCS_China %>%
+      # Add grid.region
+      left_join_error_no_match(province_names_mappings, by = "province.name") %>%
+      # Convert prices to 1990 USD per ton C and amounts to C
+      mutate(Cost_1990USDtC = Cost_2005USDtCO2 * emissions.CONV_C_CO2 / gdp_deflator(2005, 1990),
+             MtC = CO2_Mt * (1 / emissions.CONV_C_CO2)) %>%
+      arrange(province.name, Cost_2005USDtCO2) %>%
+    # Removing data for provinces with not enough onshore storage
+      filter(!(province.name %in% c("Fujian", "Guangdong", "Guangxi", "Qinghai")))
+
+    province_data <- L161.Csupply_province %>%
```

Comment:
Seems no need to generate the tibble `province_data` - just keep piping to the next section